pingcap/tiflash
# PageStorage: Fix pages are not deleted under some cases

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/5054

Problem Summary: as the issue described

### What is changed and how it works?

* StoragePool only need to remove DTFiles ids that are not exist in the V3.data, cause we perform force transform from V2.data to V3.data when `restore`
* Keep tombstone when merging the delta version for page entries in V2

### Check List

Tests 


* Manual test steps:
  * Deploy cluster and load some data into TiFlash with storage.format_version = 3
  * Change the storage.format_version = 4
  * Use `alter table t compact tiflash replica` to perform delta-merge
  * Check that the old DTFiles are removed


Side effects


Documentation


### Release note



```release-note
None
```
